public/app/features/dashboard/custom_time_ranges/TimeRangeList.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { PureComponent } from 'react';
+
+const width1 = {
+  width: '1%',
+};
+
+const lightBlue = {
+  color: '#32d1df',
+};
+
+const green = {
+  color: '#269242',
+};
+
+export interface Props {
+  ranges: any;
+  up: any;
+  down: any;
+  edit: any;
+  delete: any;
+  add: any;
+}
+
+export default class List extends PureComponent<Props> {
+  constructor(props) {
+    super(props);
+    this.state = { show: false };
+  }
```

Comment:
You can use class properties for the state.

```
export default class List extends PureComponent<Props> {
    state = { show: false }
....
```

Reply:
Thanks for tip. I removed that state completely, didn't need it after all.